tests/console/yast2_samba.pm

```diff
@@ -230,9 +227,45 @@ sub run {
     send_key 'alt-o';
 
     wait_serial('yast2-samba-server-status-0', 60) || die "'yast2 samba-server' didn't finish";
+}
+
+sub gui_new {
+    assert_screen 'yast2_samba-workgroup';
+    send_key 'alt-n';
+    if (check_screen 'yast2_still_susefirewall2', 10) {
+        record_soft_failure 'bsc#1064405';
+        send_key 'alt-c';
+    }
+    wait_still_screen;
+    # from here we have new workflow because of changes in sles 15
+}
+
+
+sub run {
+    setup_ldap;
+
+    # install samba stuffs at first
+    zypper_call("in samba yast2-samba-server");
+
+    # start samba server configuration
+    script_run("yast2 samba-server; echo yast2-samba-server-status-\$? > /dev/$serialdev", 0);
+    # check Samba-Server Configuration got started
+    if ((is_tumbleweed) || ((is_leap) && leap_version_at_least('15.0')) || (is_sle && sle_version_at_least('15'))) {
+        gui_new;
+    }
+    # continue current workflow for sle 12
+    else {
+        gui_current;
+    }
```

Comment:
Better use the same approach that you did further up by only looking for the specific older versions. This allows to not explicitly repeat the `is_tumbleweed` check

Reply:
okay, will change like:

if (((is_leap) && !(leap_version_at_least('15.0'))) || (is_sle && !(sle_version_at_least('15')))) {
        gui_current;
    }
    # continue current workflow for sle 12
    else {
        gui_new;
    }

Reply:
leap_version_at_least returns false in case it's not leap, so is_leap part is not really required